src/lib/swapper/swappers/ArbitrumBridgeSwapper/getTradeQuote/getTradeQuote.ts

```diff
@@ -0,0 +1,227 @@
+import { Erc20Bridger, EthBridger, getL2Network } from '@arbitrum/sdk'
+import type { L1ToL2TransactionRequest } from '@arbitrum/sdk/dist/lib/dataEntities/transactionRequest'
+import { ethAssetId, ethChainId, fromAssetId } from '@shapeshiftoss/caip'
+import { isEvmChainId } from '@shapeshiftoss/chain-adapters'
+import type {
+  GetEvmTradeQuoteInput,
+  SingleHopTradeQuoteSteps,
+  TradeQuote,
+} from '@shapeshiftoss/swapper'
+import {
+  makeSwapErrorRight,
+  type SwapErrorRight,
+  SwapperName,
+  TradeQuoteError,
+} from '@shapeshiftoss/swapper'
+import { KnownChainIds } from '@shapeshiftoss/types'
+import type { Result } from '@sniptt/monads'
+import { Err, Ok } from '@sniptt/monads'
+import { getDefaultSlippageDecimalPercentageForSwapper } from 'constants/constants'
+import { BigNumber } from 'ethers5'
+import { v4 as uuid } from 'uuid'
+import { getAddress, isAddressEqual } from 'viem'
+import { arbitrum } from 'viem/chains'
+import { usdcAssetId } from 'components/Modals/FiatRamps/config'
+import { bn } from 'lib/bignumber/bignumber'
+import { getEthersV5Provider } from 'lib/ethersProviderSingleton'
+import { assertGetEvmChainAdapter, getFees } from 'lib/utils/evm'
+
+import { assertValidTrade } from '../utils/helpers'
+
+const usdcOnArbitrumAssetId = 'eip155:42161/erc20:0xaf88d065e77c8cc2239327c5edb3a432268e5831'
+
+const fetchTokenFallbackGasEstimates = () =>
+  // https://github.com/OffchainLabs/arbitrum-token-bridge/blob/d17c88ef3eef3f4ffc61a04d34d50406039f045d/packages/arb-token-bridge-ui/src/util/TokenDepositUtils.ts#L45-L51
+  // Use hardcoded gas estimate values
+  // Values set by looking at a couple of different ERC-20 deposits
```

Comment:
That sucks but we can't use the SDK for ERC20 deposits - if an approval is required, Tx simulation will revert and we will be very sad.

Reply:
while they are close, we may want to use the appropriate fallback gas estimations per chain. (240000 for eth, 275000 for arb). if we use the retryable gas override 30% increase we can also take into account that and estimate high

Reply:
also looks like withdraw has some different fallback values we want to take into consideration?

Reply:
There should be no fallback needed for Arbitrum -> Ethereum tokens bridging since ERC20 *withdraws* do not require an approval, hence estimation works as-is without needing to fallback. This is strictly for ERC20 deposits, which may require an approval, and for which we can't just simulate a deposit Tx (meaning we can't consume the SDK methods, which effectively do simulate a deposit Tx) as it would revert

Reply:
Ok, I was looking here: https://github.com/OffchainLabs/arbitrum-token-bridge/blob/d17c88ef3eef3f4ffc61a04d34d50406039f045d/packages/arb-token-bridge-ui/src/util/WithdrawalUtils.ts#L90 and saw the fallback, but may just be fallback on network request fail versus the approval scenario.

Reply:
Agreed to always use fallbacks when fees are no worky e.g not enough balance meaning we can't estimate gas

Reply:
Actually going to keep things simple and only have fallback in the case of ERC20 deposits, which is the only case a user may have balance but estimation would revert. Lot of guesstimation work, and Arbitrum bridge themselves simply don't return a quote/estimated gas when there is insufficient balance
<img width="697" alt="image" src="https://github.com/shapeshift/web/assets/17035424/e9fef92c-e90d-48c2-99bc-4bcd88a575b3">